.github/workflows/publish.yml

```diff
@@ -0,0 +1,51 @@
+name: Publish
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🔒​ Check actor permission level
+        uses: skjnldsv/check-actor-permission@v2
+        with:
+          require: admin
+
+      - name: 📥​ Checkout
+        uses: actions/checkout@v3
+
+      - name: 🪄 Read package.json node and npm engines version
+        uses: skjnldsv/read-package-engines-version-actions@v1.2
+        id: versions
+        with:
+          fallbackNode: '^16'
+          fallbackNpm: '^8'
+
+      - name: 🔧​ Setup Node (NPM Registry)
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ steps.versions.outputs.nodeVersion }}
+          cache: 'npm'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: 📦 Install dependencies & build
+        run: |
+          npm ci
+          npm run build
+
+      - name: 📣 Publish package on NPM
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: 🔧​ Setup Github Registry
+        uses: actions/setup-node@v3
+        with:
+          registry-url: 'https://npm.pkg.github.com'
```

Comment:
is that where we publish?

Reply:
Now we publish only in NPM
But all major npm front-ends ([NPM](https://docs.npmjs.com/cli/v8/commands/npm-install), [Yarn](https://yarnpkg.com/cli/add), [pnpm](https://pnpm.io/cli/add#install-from-git-repository)) are able to use not only other registries, also github repos directly.

This movement is important to improve the security and resilience of the javascript ecosystem.

Reply:
Is it exclusive? Like does that means we'll only publish to the github registry instead of npm then?

Reply:
No, npm is done in the previous step https://github.com/nextcloud/nextcloud-event-bus/pull/490/files/66382a3dfcf32553b70009e9a5ad5eb1f8e00511#diff-551d1fcf87f78cc3bc18a7b332a4dc5d8773a512062df881c5aba28a6f5c48d7R38-R41

Reply:
Perfect then